v2/tools/generator/internal/config/group_configuration.go

```diff
@@ -43,15 +43,20 @@ func NewGroupConfiguration(name string) *GroupConfiguration {
 // of the version, so we can do lookups via TypeName. All indexing is lower-case to allow case-insensitive lookups (this
 // makes our configuration more forgiving).
 func (gc *GroupConfiguration) add(version *VersionConfiguration) {
-	pkg := astmodel.CreateLocalPackageNameFromVersion(version.name)
+	// Convert version.name into a package version
+	// We do this by constructing a local package reference because this avoids replicating the logic here and risking
+	// inconsistency if things are changed in the future.
+	local := astmodel.MakeLocalPackageReference("prefix", "group", astmodel.GeneratorVersionPrefix, version.name)
+	_, lv, _ := local.GroupVersion()
```

Comment:
Should we panic rather than discard here? And below?

Reply:
Don't need to - the only reason it would fail is if it was an external package reference; and we just created it, so we know that's not the case.

Reply:
Added, though I don't see how it would ever fire.

Reply:
Yeah agreed, neither do I. It pays to be paranoid sometimes though - who knows in the future when "That cannot possibly happen" can actually start happening because of some other refactor.

Thanks for adding!

Reply:
Reactivating as I don't think you actually did this at this location? (You did it below though I see)